Library/SPIClient.m

```diff
@@ -274,13 +274,7 @@ - (void)pairingCancel {
     if (_state.flow != SPIFlowPairing || _state.pairingFlowState.isFinished) return;
     
     if (_state.pairingFlowState.isAwaitingCheckFromPos && !_state.pairingFlowState.isAwaitingCheckFromEftpos) {
-        SPIMessage *message = [[[SPIDropKeysRequest alloc] init] toMessage];
-        
-        __weak __typeof(& *self) weakSelf = self;
-        
-        dispatch_async(self.queue, ^{
-            [weakSelf send:message];
-        });
+        [self send:[[[SPIDropKeysRequest alloc] init] toMessage]];
```

Comment:
Why are you no longer doing this asynchronously? What changed?

Reply:
The problem is as follows.

When pairing with a terminal from an iOS library, if you confirm pairing from the POS but rejected from the terminal, the terminal shows a disconnected status when it should unpair itself from the terminal.

The application was crashed when this situation occurs.  Terminal disconnected before unpaired and crashed the app.

Reply:
OK, fair enough. Just checking that it was an intentional change.